presto-hive-common/src/main/java/org/apache/hadoop/fs/HadoopExtendedFileSystemCache.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.fs;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+
+public class HadoopExtendedFileSystemCache
```

Comment:
Maybe I'm missing something, but could you explain the motivation for this class and the introduction of reflection?

Reply:
These are create to use our own PrestoExtendedFileSystemWrapper which is based on ExtendedFileSystem. Have a look at HadoopFileSystemCache.